webapp/channels/src/components/searchable_channel_list.tsx

```diff
@@ -0,0 +1,481 @@
+                        />
+                    </Menu>
+                </MenuWrapper>
+            );
+        }
+
+        const hideJoinedButtonClass = classNames('get-app__checkbox', {checked: this.props.rememberHideJoinedChannelsChecked});
+        const hideJoinedPreferenceCheckbox = (
+            <div
+                id={'hideJoinedPreferenceCheckbox'}
+                onClick={this.handleChecked}
+            >
+                <button
+                    className={hideJoinedButtonClass}
+                    aria-label={this.props.rememberHideJoinedChannelsChecked ? localizeMessage('more_channels.hide_joined_checked', 'Hide joined channels checkbox, checked') : localizeMessage('more_channels.hide_joined_not_checked', 'Hide joined channels checkbox, not checked')
+                    }
+                >
+                    {this.props.rememberHideJoinedChannelsChecked ? <CheckboxCheckedIcon/> : null}
+                </button>
+                <FormattedMessage
+                    id='more_channels.hide_joined'
+                    defaultMessage='Hide Joined'
+                />
+            </div>
+        );
+
+        let channelCountLabel;
+        if (channels.length === 0) {
+            channelCountLabel = localizeMessage('more_channels.count_zero', '0 Results');
+        } else if (channels.length === 1) {
+            channelCountLabel = localizeMessage('more_channels.count_one', '1 Result');
+        } else if (channels.length > 1) {
+            channelCountLabel = localizeAndFormatMessage(t('more_channels.count'), '0 Results', {count: channels.length});
+        } else {
+            channelCountLabel = localizeMessage('more_channels.count_zero', '0 Results');
+        }
+
+        const dropDownContainer = (
+            <div className='more-modal__dropdown'>
+                <span id='channelCountLabel'>{channelCountLabel}</span>
+                <div id='modalPreferenceContainer'>
+                    {channelDropdown}
+                    {hideJoinedPreferenceCheckbox}
+                </div>
+            </div>
+        );
+
+        return (
+            <div className='filtered-user-list'>
+                {input}
+                {dropDownContainer}
+                <div
+                    role='application'
+                    className='more-modal__list'
+                    tabIndex={-1}
+                >
+                    <div
+                        id='moreChannelsList'
+                        tabIndex={-1}
+                        ref={this.channelListScroll}
+                    >
+                        {listContent}
+                    </div>
+                </div>
+                <div className='filter-controls'>
+                    {previousButton}
+                    {nextButton}
+                </div>
+            </div>
+        );
+    }
+}
+
+// SearchableChannelList.defaultProps = {
```

Comment:
I really hoped since this was a new component we could have written this in functional way but it's fine either ways. Also do you plan to remove this ?